---
layout: post
title:  "LEfSe"
author: JB
date:   2019-05-16
categories: Statistics
tags:	Amplicon Metagenome LEfSe LDA
cover:  "/assets/instacode.png"
---

'*This post explains what LEfSe is and how LEfSe works. (in personal opinion and review)*'

## LEfSe ?

LDA Effect Size (LEfSe) (Segata et. al 2010) is An algorithm for High-Dimensional biomarker discovery and explanation that identifies genomic features (genes, pathways, or taxa) characterizing the differences between two or more biological conditions (or classes).

--------------------------------
## Non-paramteric data

Normal distribution does not fit well to the metagenomic data ([https://doi.org/10.21775/cimb.024.017](https://doi.org/10.21775/cimb.024.017)).

The specific property of metagenomic data is that the normal distribution implied in many statistical tests does not describe this type of data well. 
* First of all, normal distribution is continuous while the metagenomic sequencing data have discrete nature.
* Second, components of a feature vector cannot be negative. At the same time the mean value of taxon abundance is ofen comparable to its variance and the abundance matrix is sparse (has many zero values).

For example, in the case of gut microbiota, due to the wide diversity of microbial community structures, a typical bacterial species has zero or low abundance in microbiota of the majority of people while only a small fraction of the population carry high abundance (tens of per cent) of this species.A typical histogram of taxon abundance across samples is presented in Figure as below.

<a href="/assets/lefse/lefse_non-parametric.PNG" data-lightbox="falcon9-large" data-title="example">
  <img src="/assets/lefse/lefse_non-parametric.PNG" title="example">
</a>

--------------------------------
## Hypothesis testing procedures
 Hypothesis testing is a statistical method that is used in making statistical decisions using experimental data.
 The following procedure shows that how we test experiments with statistical method according to sample charateristics.
<a href="/assets/lefse/hypothesis_testing_procedures.png" data-lightbox="falcon9-large" data-title="hypothesis testing procedures">
  <img src="/assets/lefse/hypothesis_testing_procedures.png" title="Hypothesis Testing Procedures">
</a>

--------------------------------
## Normality test
First of all, we are able to confirm whether data shows normality or not throughout *Shapiro-Wilk* test.

#### Shapiro-Wilk test
The null-hypothesis of this test is that the population is normally distributed.


```r
> library(dplyr)
> library(MASS)
> library(ggplot2)

# test dataset : CO2
# The CO2 data frame from an experiment on the cold tolerance of the grass species Echinochloa crus-galli.
# Type : a factor with levels Quebec Mississippi giving the origin of the plant
# uptake : a numeric vector of carbon dioxide uptake rates (umol/m^2 sec).


#shapiro test
> summarise(group_by(CO2, Type), shapiro.pval=shapiro.test(uptake)$p.value)
# A tibble: 2 x 2
  Type        shapiro.pval
  <fct>              <dbl>
1 Quebec          0.000111
2 Mississippi     0.0213  


> ggplot(data=CO2, aes(x=uptake, fill=Type)) +
  geom_histogram(bins=10, alpha=0.5) +
  theme_bw() +
  facet_wrap(.~Type)
```


<a href="/assets/lefse/normality.png" data-title="shapiro-wilk test">
  <img src="/assets/lefse/normality.png" title="shapiro-wilk test">
</a>

--------------------------------
## Kruskal Wallis test
If data does not show normality, mean value cannot be used for comparing two or more groups.

The [Kruskal–Wallis test](https://en.wikipedia.org/wiki/Kruskal%E2%80%93Wallis_one-way_analysis_of_variance) by ranks, Kruskal–Wallis H test (named after William Kruskal and W. Allen Wallis), or one-way ANOVA on ranks is a non-parametric method for testing whether samples originate from the same distribution. It is used for comparing two or more independent samples of equal or different sample sizes. It extends the Mann–Whitney U test, which is used for comparing only two groups. The parametric equivalent of the Kruskal–Wallis test is the one-way analysis of variance (ANOVA).

* Null hypothesis: Null hypothesis assumes that the samples (groups) are from identical populations.
* Alternative hypothesis: Alternative hypothesis assumes that at least one of the samples (groups) comes from a different population than the others.

```r
> str(CO2)
 Classes ‘nfnGroupedData’, ‘nfGroupedData’, ‘groupedData’ and 'data.frame': 84 obs. of  5 variables:
 $ Plant    : Ord.factor w/ 12 levels "Qn1"<"Qn2"<"Qn3"<..: 1 1 1 1 1 1 1 2 2 2 ...
 $ Type     : Factor w/ 2 levels "Quebec","Mississippi": 1 1 1 1 1 1 1 1 1 1 ...
 $ Treatment: Factor w/ 2 levels "nonchilled","chilled": 1 1 1 1 1 1 1 1 1 1 ...
 $ conc     : num  95 175 250 350 500 675 1000 95 175 250 ...
 $ uptake   : num  16 30.4 34.8 37.2 35.3 39.2 39.7 13.6 27.3 37.1 ...

> kruskal.test(uptake ~ Type, data = CO2)

        Kruskal-Wallis rank sum test

 data:  uptake by Type
 Kruskal-Wallis chi-squared = 29.491, df = 1, p-value = 5.617e-08
```

P-value is **5.617e-08**

It means that `carbon dioxide uptake rates` have different distribution between Québec and Mississippi.



